routers/repo/pull.go

```diff
@@ -991,3 +993,40 @@ func CleanUpPullRequest(ctx *context.Context) {
 
 	ctx.Flash.Success(ctx.Tr("repo.branch.deletion_success", fullBranchName))
 }
+
+// DownloadPullDiff render a pull's raw patch
+func DownloadPullDiff(ctx *context.Context) {
+
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Handle(404, "GetIssueByIndex", err)
+		} else {
+			ctx.Handle(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	// Redirect elsewhere if it's not a pull request
+	if !issue.IsPull {
+		ctx.Redirect(ctx.Repo.RepoLink + "/issues/" + com.ToStr(issue.Index))
```

Comment:
It should not redirect as this will be confusing when using from command line to download patch, instead it should return 404

Reply:
Done with 9149d562